src/Chapter2.hs

```diff
@@ -841,7 +858,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+
+rotate :: Int -> [a] -> [a]
+rotate num l = if num < 0 
+  then []
+  else take (length l) (drop num (cycle l)) 
```

Comment:
Unfortunately, `cycle` won't work on an empty list, so this case needs to be considered separately 🙂